pipeline_dp/budget_accounting.py

```diff
@@ -1,20 +1,91 @@
 """Privacy budget accounting for DP pipelines."""
 
 
+class Budget:
+  """Manages the budget allocated for an operation."""
+
+  def __init__(self, weight, use_eps, use_delta, eps=None, delta=None):
+    """Constructs a Budget.
+
+    The values for eps and delta will be computed only after the pipeline was
+    constructed and the method compute_budgets was called.
+
+    Args:
+      weight: The weight is used to get more accurate results.
+      use_eps: A boolean that is False when the operation doesn't need epsilon.
+      use_delta: A boolean that is False when the operation doesn't need delta.
+    """
+    self._eps = eps
+    self._delta = delta
+    self.weight = weight
+    self.use_eps = use_eps
+    self.use_delta = use_delta
+
+  @property
+  def eps(self):
+    """eps: Optional; Parameter of (eps, delta)-differential privacy."""
+    if self._eps is None:
+      raise AssertionError("Privacy budget is not calculated yet.")
+    return self._eps
+
+  @property
+  def delta(self):
+    """delta: Optional; Parameter of (eps, delta)-differential privacy."""
+    if self._delta is None:
+      raise AssertionError("Privacy budget is not calculated yet.")
+    return self._delta
+
+  def set_eps_delta(self, eps, delta):
+    self._eps = eps
+    self._delta = delta
+
+
 class BudgetAccountant:
-  """Manages privacy budget."""
+  """Manages the privacy budget."""
 
   def __init__(self, epsilon, delta):
-    """Construct a BudgetAccountant
+    """Constructs a BudgetAccountant.
 
     Args:
-      epsilon, delta: parameters of (epsilon, delta)-differential privacy.
+      epsilon, delta: Parameters of (epsilon, delta)-differential privacy.
     """
     if epsilon <= 0:
-      raise ValueError(f"Epsilon must be positive, not {epsilon}")
+      raise ValueError(f"Epsilon must be positive, not {epsilon}.")
     if delta < 0:
-      raise ValueError(f"Delta must non-negative, not {delta}")
+      raise ValueError(f"Delta must be non-negative, not {delta}.")
+
     self._eps = epsilon
     self._delta = delta
+    self._requested_budgets = []
+
+  def request_budget(self, weight, use_eps, use_delta):
```

Comment:
Looking at tests, I realized that it's might be hard to distinguish use_eps/use_delta if they are used as positional arguments, please make them keyword arguments (i.e. enforce to use them to be keyword arguments), namely:
def request_budget(self, weight, *, use_eps, use_delta):